content/tutorials/MQTT-snake-js.textile

```diff
@@ -0,0 +1,255 @@
+  const mqtt = require('mqtt');
+  const keypress = require('keypress');
+```
+
+"See this step in Github":https://github.com/ably/tutorials/TODO
+
+h2. Step 4 - Detect key presses
+
+Now it's time for you to detect key presses on the keyboard. In this tutorial, you'll be needing to know when the arrow keys or return key is pressed, and then through MQTT communicate which key was pressed. You'll be detecting the key presses with the keypress NPM module. Add the below code after the library instancing in the @controller.js@ file:
+
+```[javascript]
+  keypress(process.stdin);
+
+  process.stdin.setRawMode(true);
+
+  process.stdin.on('keypress', function (ch, key) {
+    if (key) {
+      console.log('Key clicked: ' +  key.name);
+    }
+  });
+```
+
+There are three main things occuring in this code. The first is @keypress(process.stdin)@, which will simply cause our @process.stdin@ to emit @keypress@ events.
+
+Secondly is @process.stdin.setRawMode(true)@, which ensures your keyboard input is available character-by-character with no modifiers. For example, when typing ctrl-c, @process.stdin@ will output the following as a @keypress@ event: 
+
+```
+  { name: 'c',
+    ctrl: true,
+    meta: false,
+    shift: false,
+    sequence: '\u0003' 
+  }
+```
+
+Finlly, @process.stdin.on()@ is simply a listener for the @keypress@ event. You use it in this case to print to the console whenever a key is pressed. You can try this out for yourself by running the following line of code in the terminal:
+
+```[sh]
+  node controller.js
+```
+
+If all is well, you should see a message saying the key you pressed whenever you click a key.
+
+"See this step in Github":https://github.com/ably/tutorials/TODO
+
+h2. Step 5 - Connect to Ably through MQTT
+
+Now that you're able to detect key presses, it's time to start transmitting these presses through MQTT to Ably. When using MQTT with Ably, there are a few requirements with regards to your setup. You'll need to connect to 'mqtt.ably.io' on port 8883, which requires the use of SSL/TLS in your connection. If you have a device which cannot support SSL, you'll need to connect via port 1883 instead, but this will come with a number of "restrictions":https://support.ably.io/solution/articles/3000074906-using-the-mqtt-protocol-adapter. 
+
+You will need to set the keep alive time to a value between 15 and 60 seconds, where 60 seconds maximises the battery life, and 15 seconds will maximise responsiveness to network issues. In this tutorial you'll be setting it to 15.
+
+For authentication, you'll need to either use an API Key, or token. For simplicity in this tutorial you'll be using the API Key you got in step 1. To provide the API Key, you need to provide the first half of the API Key (before the colon) as the username, and the second half of the API Key (after the colon) as the password. For example, if your API key is @A12B3C.4D:5E6F7G8H@, your username will be @A12B3C.4D@ and your password will be @5E6F7G8H@.
```

Comment:
Maybe less confusing to just say 'For this, you'll need to provide the first half of the API Key...'